### ISSUE TITLE
Transfer Learning Toolkit

### ISSUE THREAD

**Issue by turowicz** [triton-inference-server/server#1625]
I'm looking into migrating from Tensorflow to Transfer Learning Toolkit.

The migration from TF-Serving to TRTIS was very beneficial, so I'm quite keen on doing the same for training and model management.

Problem is, I can't find any resource on how to get the `.hdf5` file and spin it up in TRTIS in order to see how the inference performs.

Could you guys point me to any docs on how it's done? In TRTIS docs I can only see TRT, ONNX, TF and PyTorch.

cc @deadeyegoodwin

**Comment by turowicz** [comment]
I've found the following Keras-To-ONNX approach:

https://devblogs.nvidia.com/speeding-up-deep-learning-inference-using-tensorflow-onnx-and-tensorrt/

**Comment by turowicz** [comment]
It does feel like the wrong approach though, I should be looking at TensorRT conversion.

**Comment by GuanLuo** [comment]
Right, you would need to convert it to one of the supported framework. I am not familiar with Transfer Learning Toolkit (TLT), I assume `.hdf5` is the generated file from TLT? From this [post](https://medium.com/dataseries/build-and-deploy-accurate-deep-learning-models-for-intelligent-image-and-video-analytics-8ad755213c06), it seems like TLT has an exporter to convert it to TensoRT plan

**Comment by turowicz** [comment]
Apologies @GuanLuo, I assumed that you guys work in Nvidia and know TLT well. `.hdf5` is a keras format in which the TLT guys publish their models. It can be converted to ONNX, but it cannot be then converted to TensorRT as it uses INT64 which is not supported.

I grabbed TLT assuming that surely it would play with well with the Nvidia stack, but it doesn't actually.

**Comment by JulianOrteil** [comment]
@turowicz 

Considering that .hdf5 is a Keras format you'll most likely need to convert the model to a savedmodel for the best chance at achieving what you need. From there you can convert it to a TRT model, or let the server do that on model warmup using the optimization parameter in the model config.

See https://www.tensorflow.org/guide/keras/save_and_serialize for conversion.

I use the following optimization setting in my config:
```
optimization {
    execution_accelerators {
        gpu_execution_accelerator: [{
            name: "tensorrt"
        }]
    }
}
```

**Comment by okdimok** [comment]
> but it cannot be then converted to TensorRT as it uses INT64 which is not supported.

I think the supported approach with TLT 2.0 is to fine-tune it on your own data, specifying '.hdf5' file as  `pretrained_model_path: "/path/to/your/pretrained/model"`. Training output is '.tlt' file, which can than be converted to TRT '.engine' using `tlt-export` (see [TLT docs, exporting models](https://docs.nvidia.com/metropolis/TLT/tlt-getting-started-guide/#exporting_models)). This whole pipeline can be found in `/workspace/examples/faster_rcnn/faster_rcnn.ipynb` Jupyter notebook inside the TLT container. 

The resulting engine file can be used with Triton.

**Comment by turowicz** [comment]
I will have a look thanks @okdimok

**Comment by dzier** [comment]
Please check out Model Navigator: https://github.com/triton-inference-server/model_navigator. It is a new tool that helps facilitate model conversions.
